### DISCUSSION TITLE
Why does retryResponseCodes not include SYSTEM_BUSY?

### DISCUSSION THREAD

**Discussion by tongtaodragon** [apache/rocketmq#7685] [Q&A] [answered]
Sometimes send message synchronous report error, the response code is SYSTEM_BUSY, client will not retry. In our environment we have several master broker, even happened SYSTEM_BUSY, we think client can retry automatically. We want to add SYSTEM_BUSY in retryResponseCodes. Before that, we would like to know the reason why community doesn't put this error in retryResponseCodes.

**Comment by cserwen** [comment]
See https://github.com/apache/rocketmq/pull/5845 to find more details

**Reply by tongtaodragon** [reply]
Thanks for info! But In #5845 there is no clear answer why not add SYSTEM_BUSY in default rety code list. I take a look at commercial sdk, retry not happen only when topic not exist.

**Reply by cserwen** [reply, marked as answer]
Actually, I am also very confused about this question. I personally support adding it to the retry code. But some members think that the entire system is overloaded when SYSTEM_BUSY is returned and should not continue to be retried. This will only increase the system load.

IMO, You can take a detailed look at the #5845 and #5838 , which contain some details of the discussion.
